test/framework/src/main/java/org/elasticsearch/cluster/coordination/MockSinglePrioritizingExecutor.java

```diff
@@ -53,6 +53,12 @@ protected void afterExecute(Runnable r, Throwable t) {
         throw new KillWorkerError();
     }
 
+    // Override awaitTermination to not get blocked on termination condition
+    @Override
+    public boolean awaitTermination(long timeout, TimeUnit unit) {
+        return true;
```

Comment:
return false here? Because we have not necessarily terminated all tasks.

Reply:
Yea, both are kinda wrong but returning `false` seems like at least it's not hiding anything -> changed in https://github.com/elastic/elasticsearch/pull/37686/commits/1366efc8d83d17394f426e3969bfab3de4c251d1